src/ctap/apdu.rs

```diff
@@ -119,48 +115,118 @@ impl TryFrom<&[u8]> for APDU {
+                },
+                case_type: ApduType::Short(Case::Le1),
+            });
+        }
+        if payload.len() == 1 + (byte_0 as usize) && byte_0 != 0 {
+            // Lc is one-byte long and since the size specified by Lc covers the rest of the
+            // payload there's no Le at the end
+            return Ok(APDU {
+                header: array_ref!(header, 0, APDU_HEADER_LEN).into(),
+                lc: byte_0.into(),
+                data: payload[1..].to_vec(),
+                case_type: ApduType::Short(Case::Lc1Data),
+                le: 0,
+            });
+        }
+        if payload.len() == 2 + (byte_0 as usize) && byte_0 != 0 {
+            // Lc is one-byte long and since the size specified by Lc covers the rest of the
+            // payload with ONE additional byte that byte must be Le
+            let last_byte: u32 = (*payload.last().unwrap()).into();
+            return Ok(APDU {
+                header: array_ref!(header, 0, APDU_HEADER_LEN).into(),
+                lc: byte_0.into(),
+                data: payload[1..(payload.len() - 1)].to_vec(),
+                le: if last_byte == 0x00 { 0x100 } else { last_byte },
+                case_type: ApduType::Short(Case::Lc1DataLe1),
+            });
         }
-        // TODO: Add extended length cases
-        if apdu.case_type == ApduType::default() {
-            return Err(ApduStatusCode::SW_COND_USE_NOT_SATISFIED);
+        if payload.len() > 2 {
+            // Lc is possibly three-bytes long
+            let extended_apdu_lc: usize = BigEndian::read_u16(&payload[1..]) as usize;
+            let extended_apdu_le_len: usize = if payload.len() > extended_apdu_lc {
+                payload.len() - extended_apdu_lc - 3
+            } else {
+                0
+            };
+            if byte_0 == 0 && extended_apdu_le_len <= 3 {
+                // If first byte is zero AND the next two bytes can be parsed as a big-endian
+                // length that covers the rest of the block (plus few additional bytes for Le), we
+                // have an extended-length APDU
+                let last_byte: u32 = (*payload.last().unwrap()).into();
+                return Ok(APDU {
+                    header: array_ref!(header, 0, APDU_HEADER_LEN).into(),
+                    lc: extended_apdu_lc as u16,
+                    data: payload[3..(payload.len() - extended_apdu_le_len)].to_vec(),
+                    le: match extended_apdu_le_len {
+                        0 => 0,
+                        1 => {
+                            if last_byte == 0x00 {
+                                0x100
+                            } else {
+                                last_byte
+                            }
+                        }
+                        2 => {
+                            let le_parsed = BigEndian::read_u16(&payload[payload.len() - 2..]);
+                            if le_parsed == 0x00 {
+                                0x100
```

Comment:
If I'm correct, the value should be `0x10000`.
When `Le` is encoded on 2 or 3 bytes, a 0 is interpreted as 65 536.